veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,185 @@
+# VEP #20: Seamless TCP migration and vhost-user support with passt
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [x] (R) Enhancement issue created, linking to VEP directory in [kubevirt/enhancements] (not the initial VEP PR)
+
+## Overview
+
+This proposal outlines the enhancements implemented in passt and how KubeVirt will integrate with passt to benefit from them:
+- vhost-user protocol
+- Seamless live migration of TCP sessions
+
+## Motivation
+
+Currently, KubeVirt does not support all networking requirements expected by users:
+- Seamless network connectivity on migration
+- Observability/Service Mesh support
+- High throughput (on par with tap)
+- Support for all IP network protocols
+
+Bridge binding does not support observability or service meshes, Masquerade hides the real IP from the user, and passt currently lacks seamless migration and tap-level performance.
+
+**Most** of these gaps have been addressed in the recently released version of passt. The exception is support for non-TCP/UDP/ICMP protocols, which has not yet been implemented but is planned for future releases.
+
+To leverage these features, KubeVirt requires adaptation.
+
+## Goals
+
+- Seamless live migration of TCP connections without disconnection using passt network binding.
+- High network throughput with passt network binding.
+
+## Non Goals
+
+- Supporting network protocols other than TCP/UDP/ICMP (ARP/DNS).
+- ADressing the design passt itself.
+- Supporting CNIs that do not provide sticky IPs during migration (migration succeeds, but existing connections break).
+
+## Definition of Users
+
+- Cluster Admins
+- VM/Namespace owners
+- Guest VM users – typically unaware of their network binding
+
+## User Stories
+
+- As a Guest VM user, I want optimal network performance comparable to bridge and masquerade, ensuring my virtualized applications perform well.  
+- As a Cluster Admin, I want observability and service meshes for enhanced monitoring, encryption, and load balancing.
+- As a Cluster Admin, I require seamless VM migration to perform infrastructure upgrades without affecting business continuity.
+
+## Repos
+
+- [https://github.com/kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)
+
+## Design
+
```

Comment:
Please consider adding a paragraph about the solution using the passt binding plugin and expanding it.
Also, due to the solution being quite complex, please consider adding a block diagram.

Reply:
> Paragraph done. 

Currently, a paragraph that provides an overview / context about the solution is missing. 
It would help to get an overview of the solution prior to diving into details.

Reply:
Thank you.